### ISSUE TITLE
[autodev] Overnight: correctness baseline sweep with new determinism config

### ISSUE THREAD

**Issue by penguinwu** [penguinwu/oss-model-graph-break-corpus#84]
## Goal

Run a full correctness sweep on current torch nightly with the new full-determinism config (committed today: `set_seed(42, deterministic=True)` + `CUBLAS_WORKSPACE_CONFIG=:4096:8`). Establish the baseline divergent set so the next compiler change has something to compare against.

## Why this is a good overnight task

- **Bounded scope:** sweep runs end-to-end without intervention; final output is `correctness_results.json` + a short summary
- **Deterministic procedure:** no decisions to make mid-run — checkpointing handles crashes
- **Cost is wall-clock, not human time:** ~4–6 hours on A100, perfect overnight fit
- **Verifiable outcome:** divergence count + per-model `severity_ratio` are countable, easy to spot-check in the morning

## Procedure

1. Confirm the current corpus has fresh `eval.fullgraph_ok` markers (run `tools/query.py --status full_graph` to see the model count). The correctness pass filters specs to these.
2. Resolve a current torch nightly venv (`~/envs/torch-nightly-cu126` or rebuild via `scripts/build-nightly-from-source.sh`).
3. Launch:
   ```bash
   python3 tools/run_experiment.py correctness \
       --python ~/envs/torch-nightly-cu126/bin/python \
       --workers 4 \
       --output-dir sweep_results/experiments/correctness-baseline-YYYY-MM-DD/
   ```
4. Monitor via `sweep_watchdog.py` (or check the per-pass checkpoint file periodically).
5. On completion:
   - Tabulate results by status (match / divergence / nan_inf / shape_mismatch / errors)
   - Sort divergent rows by `severity_ratio` descending — these are the filing candidates
   - Spot-check the top 5 by re-running with the same seed to confirm they reproduce
   - Write `BASELINE-FINDINGS.md` summarizing the divergent set + severity buckets

## Acceptance

- `correctness_results.json` exists in a dated experiments dir
- `BASELINE-FINDINGS.md` exists with: pass/fail counts, top 10 divergent rows, severity histogram
- Summary posted to corpus chat (1–2 sentences max — counts + headline finding)

## Why now

- Determinism config just shipped in commit (today)
- No correctness sweep has been run end-to-end since the pass was wired up; we have zero data on the current baseline
- Will be the comparison reference for any compiler change going forward

## Out of scope

- Filing per-divergence GitHub issues — leave that for a follow-on task after Peng reviews findings
- Train-mode correctness — eval-only per current MVP

**Comment by penguinwu** [comment]
**Approach revised — implementation now in identify pass, standalone sweep deprecated for routine monitoring.**

Per Peng's direction (2026-05-01): "Make correctness checking part of any corpus run." Rather than a one-off correctness sweep, the numeric check is now baked into every identify pass — every sweep produces the divergence inventory automatically.

## Shipped

- `4aa074a` — added lightweight numeric correctness check to identify pass
- `a273aa1` — tightened to strict-determinism setup mirroring `run_correctness` (set_seed deterministic=True + HF less_flaky helpers; NOT warn_only — non-deterministic eager noise would misclassify tolerance-band flakiness as compiler divergence)
- `40778bc` — wired through analysis layer: `analyze_sweep.py` numeric section, `brief_data.py` event-driven surfacing, daily-briefing skill update, `docs/running-sweeps.md` repositioning

## What identify now produces

Every full-graph model in every sweep gets these fields on its identify result:
- `numeric_status`: `match` / `divergence` / `nan_inf_introduced` / `shape_mismatch` / `dtype_mismatch` / `skipped` / `error`
- `numeric_max_diff`, `numeric_severity_ratio`, `numeric_bitwise_equal`
- `numeric_first_divergence` (when divergent)
- `numeric_skip_reason` (when skipped — e.g. `graph_break`, `custom_compile_kwargs`)

Tolerance: HF-style fp32 atol=1e-6, rtol=1e-4 (same as `run_correctness`).

## Smoke tests (torch211/cu128)

| Model | identify status | numeric_status | max_diff |
|-------|-----------------|----------------|----------|
| BayesianDetectorModel | full_graph | match | 0.0 (bitwise equal) |
| HubertModel | graph_break | skipped (reason: graph_break) | — |

## What remains

1. **First baseline sweep with new code** — next nightly or baseline sweep on torch-nightly-cu128 produces the divergent set as a side effect. Expected outcome: divergent rows surface in `analyze_sweep.py` output AND the next-day daily brief (per the `latest_sweep_numeric.age_hours <= 24` rule). I'll launch when Peng signals.
2. **Possible coverage regression** — strict determinism may convert some prior-`full_graph` models into `eager_error` (ops with no deterministic kernel). That's explicit signal we want, not noise to hide. Will measure on first baseline.
3. **Standalone `correctness` pass** — kept for deep audits (custom backends like inductor, investigations of routine-sweep divergences). Not retired.

Closing this issue once the first baseline sweep completes and findings are summarized in a comment here.

**Comment by penguinwu** [comment]
**Correction to prior comment** — `5e1a111` reverts the brief_data + skill wiring. Per Peng's feedback (2026-05-01 PM): sweeps run on demand, not on a daily schedule, so hard-wiring sweep-specific data into the daily brief was over-engineering and would have made the brief project-specific (won't reuse when focus shifts to other projects). After-sweep analysis already surfaces regression/improvement; if a sweep happens on a given day it shows up in the brief through the existing channels (commits, closed_issues, handoff) like any other project work.

`analyze_sweep.py` keeps its numeric-correctness section — that's the proper home, invoked manually after each sweep run. Only the brief_data + skill changes were reverted.

## Open question: HF less_flaky_test helpers

Reading `set_config_for_less_flaky_test` and `set_model_for_less_flaky_test` source: they set every `LayerNorm`/`RMSNorm`/`BatchNorm` epsilon to **1.0** (instead of the model's real `1e-5` / `1e-12`). They DO NOT touch inputs — our heuristic-based bigger inputs flow through `create_model()` unchanged. Confirmed by reading + by Gate 4-mini smoke test on Blip2VisionModel (271 MB GPU), BeitModel (109 MB), ConvBertModel (180 MB) — all clean.

But eps=1.0 makes the model produce non-realistic logits. Eager-vs-compiled comparison stays valid (both runs see the same modified model), but if we'd rather see real-model numerics and accept some noise floor from low-variance norms, dropping the less_flaky helpers is reasonable. **TBD** based on Peng's preference.
